pkg/controller/versions/util.go

```diff
@@ -65,41 +65,82 @@ func checkDiscoverySelectorsNotSet(spec *v2.ControlPlaneSpec, allErrors []error)
 	return allErrors
 }
 
-func validateGlobal(ctx context.Context, version Ver, meta *metav1.ObjectMeta, spec *v2.ControlPlaneSpec, cl client.Client, allErrors []error) []error {
-	isClusterScoped, err := version.Strategy().IsClusterScoped(spec)
-	if err != nil {
-		return append(allErrors, err)
-	}
+func validateGlobal(ctx context.Context, meta *metav1.ObjectMeta, newSmcp *v2.ControlPlaneSpec, cl client.Client, allErrors []error) []error {
 	smcps := v2.ServiceMeshControlPlaneList{}
-	err = cl.List(ctx, &smcps)
+	err := cl.List(ctx, &smcps)
 	if err != nil {
 		return append(allErrors, err)
 	}
 
-	if isClusterScoped {
-		// allow SMCP create/update only when no SMCP exists or when a single SMCP exists and we're updating it
-		if len(smcps.Items) > 1 || len(smcps.Items) == 1 && smcps.Items[0].UID != meta.GetUID() {
-			return append(allErrors,
-				fmt.Errorf("a cluster-scoped SMCP may only be created when no other SMCPs exist"))
+	otherSmcpExists := fmt.Errorf("a cluster-scoped SMCP may only be created when no other SMCPs exist")
+	if newSmcp.IsClusterScoped() {
+		// an SMCP already exists and new one is created
+		if len(smcps.Items) == 1 && smcps.Items[0].UID != meta.GetUID() {
+			currentSmcp := smcps.Items[0].Spec
+			if currentSmcp.IsClusterScoped() {
```

Comment:
I suggest starting with the negative cases, so:
```golang
	if newSmcp.IsClusterScoped() {
		// an SMCP already exists and new one is created
		if len(smcps.Items) == 1 && smcps.Items[0].UID != meta.GetUID() {
			currentSmcp := smcps.Items[0].Spec
			if !currentSmcp.IsClusterScoped() {
				return append(allErrors, otherSmcpExists)
			}
```

Reply:
This condition is incorrect, because when existing SMCP is multi-tenant we have to check if the new one is cluster-wide gateway controller, so it could be:
```
	if newSmcp.IsClusterScoped() {
		// an SMCP already exists and new one is created
		if len(smcps.Items) == 1 && smcps.Items[0].UID != meta.GetUID() {
			currentSmcp := smcps.Items[0].Spec
			if !currentSmcp.IsClusterScoped() && !newSmcp.IsGatewayController() {
				return append(allErrors, otherSmcpExists)
			}
```